pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/ControllerBasedBrokerSelector.java

```diff
@@ -32,11 +33,6 @@ public class ControllerBasedBrokerSelector implements BrokerSelector {
   private final UpdatableBrokerCache _brokerCache;
   private final Properties _properties;
 
-  public ControllerBasedBrokerSelector(String scheme, String controllerHost, int controllerPort)
-      throws Exception {
-    this(scheme, controllerHost, controllerPort, new Properties());
-  }
-
```

Comment:
Why do we need to remove this constructor?

Reply:
It is not used. I did a bit of clean up as I was editing this file. Maybe I should put it back? Other forks and libraries maybe using it?

Reply:
It is just nit, but we sometimes break third party forks when deleting things like that. Given the nature of this class, I don't think that will be the case, but in general I would say it is safer to just deprecate the constructor to indicate it will be removed soon.

Anyway, nothing important. We apply these incompatibility changes all the time :D